### ISSUE TITLE
Define input roadmap

### ISSUE THREAD

**Issue by chrisforbes** [engineers-nightmare/engineers-nightmare#112]
@RobotCaleb has already improved the input system a lot. We should work out what else we need. Some ideas:

- Do we need machinery to maintain configs as the game evolves?
- Input config UI within the game
- Saving of config when edited
- Move sensitivity controls, invert, etc into config.
- Do we need to have per-context configs (flying the ship vs running around, etc)

**Comment by mkfifo** [comment]
I think we are going to definitely need context sensitive input, even for small things like being able to navigate through the menu with arrow keys.

**Comment by RobotCaleb** [comment]
> Do we need machinery to maintain configs as the game evolves?

Could you expand on what you're thinking in regards to this?

I think per-context controls makes sense. What's the scope for that? What contexts are planned?

This conversation will overlap a lot with a general config discussion.

**Comment by chrisforbes** [comment]
If you have some weird config, and we add a new action that you don't have a binding for, should we have versioning to be able to add a default for you, or just leave it unbound?

**Comment by RobotCaleb** [comment]
I think it'd be nice to model it on UE's INI merging concept. We ship a
base config, any deviations from that get serialized. First we apply base,
then apply user.

**Comment by chrisforbes** [comment]
WFM.

**Comment by mkfifo** [comment]
I like it, we carry around a changeset that we rebase onto the current stable :)

**Comment by RobotCaleb** [comment]
Some people might want to remove, for example, a key binding completely. This should count as a deviation, so we'll want some sort of "action_left" = "input_unbound" rather than just not having "action_left" defined in user.cfg as that would get overridden by base.

**Comment by mkfifo** [comment]
@RobotCaleb +1, very good point.

**Comment by RobotCaleb** [comment]
Config merging added in #130

**Comment by chrisforbes** [comment]
What's left here? Just in-game UI for binding?

**Comment by RobotCaleb** [comment]
Yes. Just that. Might require more plumbing to metadata lookups for automatically generating in game UI, but that'd be up to the implementer.
